tests/rkt_exec_test.go

```diff
@@ -43,17 +45,70 @@ func TestRunOverrideExec(t *testing.T) {
 			expectedLine: "inspect execed as: /inspect-link",
 		},
 	} {
-		child, err := gexpect.Spawn(tt.rktCmd)
-		if err != nil {
-			t.Fatalf("%d: cannot exec rkt: %v", i, err)
-		}
+		runRktAndChkOutput(t, tt.rktCmd, tt.expectedLine)
+	}
+}
+
+func TestRunPreparedExec(t *testing.T) {
+	execImage := patchTestACI("rkt-exec-override.aci", "--exec=/inspect")
+	defer os.Remove(execImage)
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	var rktCmd, uuid, expected string
+
+	// Sanity check - make sure no --exec override prints the expected exec invocation
+	rktCmd = fmt.Sprintf("%s prepare --insecure-skip-verify %s -- --print-exec", ctx.cmd(), execImage)
+	uuid = runRktAndGetUUID(t, rktCmd)
+
+	rktCmd = fmt.Sprintf("%s run-prepared --mds-register=false %s", ctx.cmd(), uuid)
+	expected = "inspect execed as: /inspect"
+	runRktAndChkOutput(t, rktCmd, expected)
+
+	// Now test overriding the entrypoint (which is a symlink to /inspect so should behave identically)
+	rktCmd = fmt.Sprintf("%s prepare --insecure-skip-verify %s --exec /inspect-link -- --print-exec", ctx.cmd(), execImage)
+	uuid = runRktAndGetUUID(t, rktCmd)
+
+	rktCmd = fmt.Sprintf("%s run-prepared --mds-register=false %s", ctx.cmd(), uuid)
+	expected = "inspect execed as: /inspect-link"
+	runRktAndChkOutput(t, rktCmd, expected)
+}
 
-		if err = expectWithOutput(child, tt.expectedLine); err != nil {
-			t.Fatalf("%d: didn't receive expected output %q: %v", i, tt.expectedLine, err)
-		}
+func runRktAndChkOutput(t *testing.T, rktCmd, expectedLine string) {
+	child, err := gexpect.Spawn(rktCmd)
+	if err != nil {
+		t.Fatalf("cannot exec rkt: %v", err)
+	}
+
+	if err = expectWithOutput(child, expectedLine); err != nil {
+		t.Fatalf("didn't receive expected output %q: %v", expectedLine, err)
+	}
 
-		if err = child.Wait(); err != nil {
-			t.Fatalf("%d: rkt didn't terminate correctly: %v", i, err)
-		}
+	if err = child.Wait(); err != nil {
+		t.Fatalf("rkt didn't terminate correctly: %v", err)
 	}
 }
+
+func runRktAndGetUUID(t *testing.T, rktCmd string) string {
```

Comment:
would you mind splitting this into a shared file (rkt_tests.go perhaps) and using it in the pod manifest test too, now? Can do that in a follow up if you like